external/fv3fit/fv3fit/_shared/stacking.py

```diff
@@ -41,8 +41,8 @@ def _stack_batch(self, ds_unstacked: xr.Dataset) -> xr.Dataset:
             dim=SAMPLE_DIM_NAME
         )
         ds = check_empty(ds)
-        ds = preserve_samples_per_batch(ds)
-        return shuffled(self._random_state, [ds])[0]
+        ds = shuffled(self._random_state, [ds])[0]
+        return preserve_samples_per_batch(ds)
```

Comment:
Could you rename `preserve_samples_per_batch` to something that describes what it's doing (subsampling the data by the inverse of the number of "dataset" coordinates)? From our discussion this function has nothing to do with uniformity of samples per batch.